backend-diagram-converter/webapp/src/main/java/org/camunda/community/converter/webapp/ConverterController.java

```diff
@@ -28,64 +30,97 @@
-      @RequestParam("files") MultipartFile[] bpmnFiles,
+      @RequestParam(value = "files", required = false) MultipartFile[] bpmnFiles,
       @RequestParam("appendDocumentation") boolean appendDocumentation) {
-    Map<MultipartFile, Exception> exceptions = new HashMap<>();
-    ByteArrayOutputStream bo = new ByteArrayOutputStream();
-    try (ZipOutputStream out = new ZipOutputStream(bo)) {
-      for (MultipartFile bpmnFile : bpmnFiles) {
-        try (InputStream in = bpmnFile.getInputStream()) {
-          BpmnModelInstance modelInstance = Bpmn.readModelFromStream(in);
-          bpmnConverter.convert(modelInstance, appendDocumentation);
-          ZipEntry entry = new ZipEntry("converted-c8-" + bpmnFile.getOriginalFilename());
-          out.putNextEntry(entry);
-          out.write(bpmnConverter.printXml(modelInstance.getDocument(), true).getBytes());
-        } catch (Exception e) {
-          exceptions.put(bpmnFile, e);
+    if (bpmnFiles != null && !engineEnabled) {
+      Map<MultipartFile, Exception> exceptions = new HashMap<>();
+      ByteArrayOutputStream bo = new ByteArrayOutputStream();
+      try (ZipOutputStream out = new ZipOutputStream(bo)) {
+        for (MultipartFile bpmnFile : bpmnFiles) {
+          try (InputStream in = bpmnFile.getInputStream()) {
+            BpmnModelInstance modelInstance = Bpmn.readModelFromStream(in);
+            bpmnConverter.convert(modelInstance, appendDocumentation);
+            ZipEntry entry = new ZipEntry("converted-c8-" + bpmnFile.getOriginalFilename());
+            out.putNextEntry(entry);
+            out.write(bpmnConverter.printXml(modelInstance.getDocument(), true).getBytes());
+          } catch (IOException e) {
+            exceptions.put(bpmnFile, e);
+          }
         }
+      } catch (IOException e) {
+        return ResponseEntity.internalServerError().body(e.getMessage());
+      }
+      if (exceptions.isEmpty()) {
+        Resource file = new ByteArrayResource(bo.toByteArray());
+        return ResponseEntity.ok()
+            .header(
+                HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"conversionResult.zip\"")
+            .contentType(MediaType.APPLICATION_OCTET_STREAM)
+            .body(file);
+      } else {
+        return ResponseEntity.badRequest()
+            .body(
+                exceptions.entrySet().stream()
+                    .map(e -> new SimpleEntry<>(e.getKey().getOriginalFilename(), e.getValue()))
+                    .collect(Collectors.toList()));
       }
-    } catch (IOException e) {
-      return ResponseEntity.internalServerError().body(e.getMessage());
     }
-    if (exceptions.isEmpty()) {
+    if (bpmnFiles == null && engineEnabled) {
+      ByteArrayOutputStream bo = new ByteArrayOutputStream();
+      try (ZipOutputStream out = new ZipOutputStream(bo)) {
+        for (Entry<String, BpmnModelInstance> bpmnFile :
+            bpmnConverter.convertFromEngine(appendDocumentation).entrySet()) {
```

Comment:
Don't read the file from the engine in the web app, it should be done in the cli module.